src/editors/data/redux/requests/reducer.test.js

```diff
@@ -0,0 +1,48 @@
+import { initialState, actions, reducer } from './reducer';
+import { RequestStates, RequestKeys } from '../../constants/requests';
+
+const testingState = {
+  ...initialState,
+  arbitraryField: 'arbitrary',
+};
+describe('requests reducer', () => {
+  test('intial state generated on create', () => {
+    expect(reducer(undefined, {})).toEqual(initialState);
+  });
+
+  describe('handling actions', () => {
+    const arbitraryKey = 'ArbItrAryKey';
+    const requestsList = [RequestKeys.fetchUnit, RequestKeys.fetchBlock, RequestKeys.saveBlock, arbitraryKey];
+
+    requestsList.forEach(requestKey => {
+      describe(`${requestKey} lifecycle`, () => {
+        const testAction = (action, args, expected) => {
+          expect(reducer(testingState, actions[action](args))).toEqual({
+            ...testingState,
+            [requestKey]: expected,
+          });
+        };
+        test('startRequest sets pending status', () => {
+          testAction('startRequest', requestKey, { status: RequestStates.pending });
+        });
+        test('completeRequest sets completed status and loads response', () => {
+          testAction(
+            'completeRequest',
+            { requestKey },
+            { status: RequestStates.completed },
+          );
+        });
+        test('failRequest sets failed state and loads error', () => {
+          testAction(
+            'failRequest',
+            { requestKey },
+            { status: RequestStates.failed },
+          );
+        });
+        test('clearRequest clears request state', () => {
+          testAction('clearRequest', { requestKey }, {});
```

Comment:
For this test, you should probably specifically give an input test state that has a non-empty state for the given request.  If you want, you could achieve this in your testAction method by calling the reducer with an initial state of ({ ...testingState, [requestKey]: { arbitrary: 'state' } });

Reasoning here is that we are importing the initial state, and this is a reasonable initial state to have expected.